### ISSUE TITLE
`/etc/keyd/default.conf` isn't an INI

### ISSUE THREAD

**Issue by zeronobody0** [NixOS/nixpkgs#345167]
### Describe the bug
[The keyd module writes keybind configurations using the INI genorator](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/services/hardware/keyd.nix), causing `=` to be escaped.
### Steps To Reproduce
1. Write a keyd config, particularly one that rebinds `=`.

### Expected behavior
The keyd config will rebind keys as stated in user configuration.

### Additional context
Was duscivered during an attempt to replicate a [symmetrical keyboard layout](https://community.frame.work/t/45150).

### Notify maintainers

@peterhoeg @woojiq

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.5, NixOS, 24.11 (Vicuna), 24.11.20240818.8a33541`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.92.0-dev-pre20240813-f9a3bf6
System type: x86_64-linux
Additional system types: aarch64-linux, i686-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/axolotl/.config/nix/nix.conf:/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/bqns0s2g37v08lqklxcq7ym0iksgfb15-lix-2.92.0-dev-pre20240813-f9a3bf6/share`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/d9gbq853jvbccrz5g3y0irbwgc57w137-source`
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by woojiq** [comment]
Use `extraConfig` as a workaround: https://search.nixos.org/options?channel=unstable&show=services.keyd.keyboards.%3Cname%3E.extraConfig&from=0&size=50&sort=relevance&type=packages&query=keyd

**Comment by woojiq** [comment]
Writing a good keyd configuration "nix way" is hard/impossible because it keyd doesn't use default file formats like toml, json, etc. It also depends on the order, so the nix language isn't very suitable for this. Just use extraConfig :)

**Comment by zeronobody0** [comment]
@woojiq
> Writing a good keyd configuration nix module (using nix best practices) is hard/impossible

Well, that's a shame. Anyway, would it be possible to use the below patch to write a working keyd configuration nix module (using nix practices of dubious quality) instead?

```diff
From 98535d1d60dae79072dcab8770899cddcaebf49b Mon Sep 17 00:00:00 2001
From: shitpostalotl <shitpostalotl@unix.dog>
Date: Sat, 28 Sep 2024 21:30:38 -0700
Subject: [PATCH 1/1] nixos/services.keyd: replace toINI program-specific gen
 fn

---
 nixos/modules/services/hardware/keyd.nix | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nixos/modules/services/hardware/keyd.nix b/nixos/modules/services/hardware/keyd.nix
index 92b62fe2e..0e25d9710 100644
--- a/nixos/modules/services/hardware/keyd.nix
+++ b/nixos/modules/services/hardware/keyd.nix
@@ -101,7 +101,7 @@ in
             [ids]
             ${lib.concatStringsSep "\n" options.ids}
 
-            ${lib.generators.toINI {} options.settings}
+            ${lib.concatLines (lib.attrsets.mapAttrsToList (section: binds: "[${section}]\n"+(lib.concatLines (lib.attrsets.mapAttrsToList (key: value: "${key} = ${value}") binds))) options.settings)}
             ${options.extraConfig}
           '';
         })
-- 
2.45.2
```

**Comment by zeronobody0** [comment]
Potential fix in #345243.

**Comment by Atemu** [comment]
Duplicate of https://github.com/NixOS/nixpkgs/issues/236622

I think the docs for the options are pretty clear about when to use which and what to do with special chars.
